components/admin_console/admin_sidebar/admin_sidebar.jsx

```diff
@@ -284,46 +278,56 @@ export default class AdminSidebar extends React.Component {
         return (
             <div className='admin-sidebar'>
                 <AdminSidebarHeader/>
-                <div className='nav-pills__container'>
-                    <Highlight filter={this.state.filter}>
-                        <ul className='nav nav-pills nav-stacked'>
-                            <li className='filter-container'>
-                                <SearchIcon
-                                    id='searchIcon'
-                                    className='search__icon'
-                                    aria-hidden='true'
-                                />
-                                <input
-                                    className={'filter ' + (this.state.filter ? 'active' : '')}
-                                    type='text'
-                                    onChange={this.onFilterChange}
-                                    value={this.state.filter}
-                                    placeholder={Utils.localizeMessage('admin.sidebar.filter', 'Find settings')}
-                                    ref={this.searchRef}
-                                />
-                                {this.state.filter &&
-                                    <div
-                                        className='sidebar__search-clear visible'
-                                        onClick={this.handleClearFilter}
-                                    >
-                                        <OverlayTrigger
-                                            delayShow={Constants.OVERLAY_TIME_DELAY}
-                                            placement='bottom'
-                                            overlay={filterClearTooltip}
+                <Scrollbars
+                    ref='scrollbar'
+                    autoHide={true}
+                    autoHideTimeout={500}
+                    autoHideDuration={500}
+                    renderThumbHorizontal={(props) => renderScrollView(props, 'scrollbar--horizontal')}
```

Comment:
We prefer to no regenerate functions in each render, do you mind to move this in independent functions like the `renderScrollView` but one per each class? I know Scrollbars is not a pure component so is going to re-render anyway, but we like to follow the policy to not generate bad examples in the code.

Reply:
done